### DISCUSSION TITLE
v1.120.0 - 50.000 Stars Release

### DISCUSSION THREAD

**Discussion by alextran1502** [immich-app/immich#13953] [Announcements]
# v1.120.0 - 50.000 Stars Release

Welcome to release `v1.120.0` of Immich. We've reached **50.000** stars on GitHub. Thank you for your love, support, and contributions to the project. 

This release introduces a new built-in automatic database dump feature as well as bug fixes and enhancements. Some of the highlights include:

- Built-in automatic database backups
- Timeline location indicator on scrolling
- The web is now on [Svelte 5](https://svelte.dev/blog/svelte-5-is-alive)
- Faster CPU transcoding for HDR videos
- Notable fix: Slow server start-up time on some systems

### Built-in automatic database backups

This release introduces a mechanism that automatically creates a database dump and is enabled by default. It will run at 2 AM server time and keep the last 14 backups, all of which are configurable. You can find the settings in [Administration > Settings > Backup Settings](https://my.immich.app/admin/system-settings?isOpen=backup)

![image](https://github.com/user-attachments/assets/a24c8fa9-9668-4e51-868a-934e0be91f0f)

The database zip files are placed at `<UPLOAD_LOCATION>/backups`. 

Below is an example of a database backup size for a library with 80_000 assets.

![image](https://github.com/user-attachments/assets/3eedd45b-8e35-42b2-929d-d11d6b5b4410)

Currently, restores must be done manually, and instructions can be found in the [documentation](https://main.preview.immich.app/docs/administration/backup-and-restore#restoring)

### Timeline location indicator on scrolling

The timeline on the web will now show the Month/Year information while scrolling to better indicate the viewport context.

https://github.com/user-attachments/assets/8453ed7c-9828-444b-8ea6-bf0f6f4d6aa5

### Faster CPU transcoding for HDR videos

Immich now uses a faster tone-mapping implementation developed by the Jellyfin team to map colors from HDR to SDR. During testing, overall transcoding speed improved by as much as 69% for a 4K target resolution. The exact improvement will depend on hardware, source video, and transcoding settings.

> [!NOTE]
> Note for third-party Immich distributions: as this filter only exists in **jellyfin-ffmpeg**, please ensure you use this build instead of a standard FFmpeg build.

## Support Immich

<p align="center">
<img src="https://media.giphy.com/media/v1.Y2lkPTc5MGI3NjExbjY2eWc5Y2F0ZW56MmR4aWE0dDhzZXlidXRmYWZyajl1bWZidXZpcyZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/87CKDqErVfMqY/giphy.gif" width="450" title="SUPPORT THE PROJECT!"> 
</p>

If you find the project helpful, you can support Immich by purchasing a product key at [https://buy.immich.app](https://buy.immich.app).

Cheers! 🍻

- - - -

### 🚀 Features
* feat(web): scrubber label and animation by @alextran1502 in https://github.com/immich-app/immich/pull/13815
* feat: built-in automatic database backups by @zackpollard in https://github.com/immich-app/immich/pull/13773
### 🌟 Enhancements
* feat: mount checks on a folder level by @zackpollard in https://github.com/immich-app/immich/pull/13801
* feat(web): library settings crontab link includes existing expression by @zackpollard in https://github.com/immich-app/immich/pull/13807
* feat(web): disable opening image and library sub-items by default by @ITestInProd in https://github.com/immich-app/immich/pull/13729
* fix(mobile): Reapply thumbnail image improvements and fix issue #13804 by @dvbthien in https://github.com/immich-app/immich/pull/13835
* feat(server): use tonemapx for software tone-mapping by @mertalev in https://github.com/immich-app/immich/pull/13785
* feat: render asset path by @cfitzw in https://github.com/immich-app/immich/pull/13873
* chore(mobile): upgrade gradle by @alextran1502 in https://github.com/immich-app/immich/pull/13901
### 🐛 Bug fixes
* fix(server): handle N/A duration response from ffprobe by @jrasm91 in https://github.com/immich-app/immich/pull/13803
* fix: shutdown api process when another worker exits unexpectedly by @zackpollard in https://github.com/immich-app/immich/pull/13802
* fix(mobile): Revert thumbnail image ui improvements  (#13655) by @alextran1502 in https://github.com/immich-app/immich/pull/13806
* fix(server): keep system config transformations by @jrasm91 in https://github.com/immich-app/immich/pull/13796
* fix(mobile): thumbnail not filled area on tablet by @alextran1502 in https://github.com/immich-app/immich/pull/13808
* fix(mobile): search page by @alextran1502 in https://github.com/immich-app/immich/pull/13833
* fix(mobile): not throwing error when cannot parse orientation value by @alextran1502 in https://github.com/immich-app/immich/pull/13853
* fix(mobile): do not removed not backup asset when selecting the correspond options by @yashrajjain726 in https://github.com/immich-app/immich/pull/13256
* fix(server): wrong image dimensions for RAW files (RAF, CR2) (also fixes face preview) by @C-Otto in https://github.com/immich-app/immich/pull/13377
* fix: healthcheck if custom host is set by @danieldietzler in https://github.com/immich-app/immich/pull/13887
* fix: remove duplicateIds on unique assets by @Pranay-Pandey in https://github.com/immich-app/immich/pull/13752
### 📚 Documentation
* chore(docs): update german readme to latest version by @MickLesk in https://github.com/immich-app/immich/pull/13824
* chore(docs): Add Immich Public Proxy to the Community Projects list by @alangrainger in https://github.com/immich-app/immich/pull/13836
* docs: fail2ban community project by @mmomjian in https://github.com/immich-app/immich/pull/13943
### 🌐 Translations
* chore(web): update translations by @weblate in https://github.com/immich-app/immich/pull/13688
* chore(web): update translations by @weblate in https://github.com/immich-app/immich/pull/13810

## New Contributors
* @ITestInProd made their first contribution in https://github.com/immich-app/immich/pull/13729
* @alangrainger made their first contribution in https://github.com/immich-app/immich/pull/13836

**Full Changelog**: https://github.com/immich-app/immich/compare/v1.119.1...v1.120.0

<hr /><em>This discussion was created from the release <a href='https://github.com/immich-app/immich/releases/tag/v1.120.0'>v1.120.0 - 50.000 Stars Release</a>.</em>

**Comment by VenomousViper27** [comment]
I realize this is probably a very simple question, but does the built in database back up mean that I can remove the following section from my docker-compose.yml file?

backup:
    container_name: immich_db_dumper
    image: prodrigestivill/postgres-backup-local
    env_file:
      - .env
    environment:
      POSTGRES_HOST: database
      POSTGRES_DB: ${DB_DATABASE_NAME}
      POSTGRES_USER: ${DB_USERNAME}
      POSTGRES_PASSWORD: ${DB_PASSWORD}
      SCHEDULE: "@daily"
      BACKUP_NUM_KEEP: 7
      BACKUP_DIR: /db_dumps
    volumes:
      - ./db_dumps:/db_dumps
    depends_on:
      - database

If I can/should remove this, do I just need to run 'sudo docker compose up -d' afterwards? Appreciate the help and any guidance!

**Reply by alextran1502** [reply]
Yeah, with the built-in backup, you can remove this service

**Reply by bcsteeve** [reply]
The documentation needs a re-work then, because there are currently two competing "Automatic Database Backups" sections on that page, and (confusingly) one section that says automatic restores aren't here yet and further down a section titled, "Automated Restore".

Very confusing.  Now that you have a built-in automatic backup scheme... just get rid of the rest.

**Reply by bo0tzz** [reply]
#13961

**Comment by jaxgun** [comment]
I've encountered an issue after update to this release. When viewing videos on the mobile app, the video starts playing at as expected but the video appears to freeze then a smaller version of the same video plays. The original video appears as a border around the playing, smaller video. Doesn't affect all files. I reinstalled the app but it didn't make a difference. This is happening on iOS and Android App. I will downgrade the server to test when I get a chance. I can provide whatever logs if needed.

Thanks for the great project! 
![Screenshot_20241106-160206](https://github.com/user-attachments/assets/f381a4bb-0d77-40f1-8b06-8bdc6a8121c3)

**Reply by bo0tzz** [reply]
Have you made any changes to the thumbnail or transcode settings on the server?

**Reply by jaxgun** [reply]
Only on initial installation, a few months ago. Have not touched any transcode or thumbnail settings directly before or after the upgrade/update today.

**Reply by jaxgun** [reply]
This does not happen when viewing with on the web, only on the apps from what I can tell. Just tested the same videos.

**Reply by alextran1502** [reply]
Does it happen on all videos? What is your phone model?

**Reply by jaxgun** [reply]
It was happening about 80% of videos. I was using the app sourced through github on obtainium using the Pixel 8 Pro. I reinstalled and that didn't fix it. I just installed the Google Play version and it appears to be working properly. Maybe an issue with installing through obtainium. 

I'll consider this solved. Unless you would like me to test it further.

**Reply by jaxgun** [reply]
Thanks for taking the time, apologies for an inconvenience.

**Reply by jaxgun** [reply]
One thing to note, the Google Play version appears to be version 1.119.0 build.164. The obtainium version was 1.120.0.

**Reply by alextran1502** [reply]
It is probably a bug with the player on the new version. We did have a version upgrade for the player in the last release.

**Reply by alextran1502** [reply]
Yup, confirmed a bug

**Reply by mouie** [reply]
Hi team, chiming in here. I'm also on a Pixel 8 Pro, just updated to v1.120.0 on server and mobile, and have discovered both the above issue, but also on older videos from external library, a horizontal green bar at the bottom of every video instead. Hopefully also a bug in the new video player, so it can be squashed 😅

![Screenshot_20241107-224126_0.jpg](https://github.com/user-attachments/assets/5818f59e-7d08-402f-a66d-3aa8be387eb5)

**Reply by mertalev** [reply]
There's a good chance we won't be using this video player in the next release 😉

**Comment by ktruo010** [comment]
Would this backup feature also create backups for external libraries or just assets uploaded via Immich?

**Reply by Banana-Bond** [reply]
The database contains information of all the processed assets in our photo library.  So I am guessing it will create backups for both.

**Comment by DeuSCaNoN** [comment]
If I wanted more data safety, and have my backup on a separate disk, is it possible to mount the backup location like this in the docker compose?
```
volumes:
      - ${UPLOAD_LOCATION}:/usr/src/app/upload
      - ${BACKUP_LOCATION}:/usr/scr/app/upload/backup
      - /etc/localtime:/etc/localtime:ro
```
Where I have an env variable pointing to a file path to my 2nd hard drive?

Edit: For anyone seeing this in the future, I wouldn't recommend doing this. IT DOES work, but docker is pretty flimsy on how it handles volumes that overlap like this.

I noticed that my ${UPLOAD_LOCATION} HAD a backups folder, and my ${BACKUP_LOCATION} had the `.immich` file. If I deleted the backups folder in ${UPLOAD_LOCATION} it would delete the backups folder within the container and basically unmount the ${BACKUP_LOCATION.

Unless you know docker volume mounting well, probably wait until a feature extension where the backup location is customizable.

**Reply by jrasm91** [reply]
Yup, that should work fine. It'd be easy enough to verify as well

**Reply by DeuSCaNoN** [reply]
If that's the case, a nice chore commit can be to document the customization of the backup directory in the README.md
and thanks for the quick reply.

**Reply by jrasm91** [reply]
Feel free to open a PR

**Reply by DeuSCaNoN** [reply]
I'm busy tonight, if no one has made a PR when I test it, then I'll create a PR

**Reply by chenyanzhe** [reply]
It will be great if we can update the guide [here](https://immich.app/docs/guides/custom-locations).

**Reply by DeuSCaNoN** [reply]
@jrasm91 so I tested it, and it DOES work, but man does docker feel finicky when you have 2 volumes mounted like this.

I noticed both my `/mnt/hdd1/immich/library` had a backups folder as well as my /mnt/hdd2/ mount point. If I deleted the backups folder in my hdd1, it would completely remove the backup folder from within the container.

I'm not sure it's recommended for users that aren't familiar with this to use this setup in my opinion

**Comment by ghan1t** [comment]
Thank you for adding an automatic db backup feature to immich! This will help so many people recover from a broken update!

**Comment by Freeben666** [comment]
Thank you for this release. I don't know if it's related toi the move to Svelte 5, but the UI feels "snappier".

One small semantic comment: I'd call the new feature an automatic DB **dump**, not automatic DB backup, since it's not really backed up. A backup would be taking this DB dump and storing it somewhere else where it would be safe in case something happened to the machine running Immich.
This can be misleading for some users who might get the feeling their DB is backed up and safe, when in fact, if anything serious happens to their Immich server, they'll lose their dumps alongside the rest.

**Comment by j-seixas** [comment]
Hey! Does the db dump backup still follow the same pattern as it was before (with the added config to the compose file)? With having backups not only for daily, but still keeping weekly and monthly backup stored as well?

**Reply by zackpollard** [reply]
No, backups are currently just "keep last x", so if you backup daily it'll keep the last 14 days.

**Reply by j-seixas** [reply]
Don't like that so much as I may not notice something going wrong and want to recover from older date. Although a simple script can achieve the same thing for storing weekly/monthly from those daily ones 
But while I don't make the script, is the dump generated basically the same as using from the previous method of the dump backup? If so I can keep using the other method and if I need to restore I can follow the guide in the docs?

**Comment by Ender-Wang** [comment]
Would restoring db backups via GUI in Settings be a thing soon? #5441
